src/components/Dropdown/Dropdown.tsx

```diff
@@ -74,6 +74,7 @@ export function Dropdown(props: PropsWithChildren<DropdownProps>): JSX.Element {
   }, [parentQuery, lastTypedOrSubmittedValue, updateFocusedItem, setLastTypedOrSubmittedValue]);
 
   useRootClose(containerRef, () => {
+    onSelect?.(value, focusedIndex, focusedItemData);
```

Comment:
This change improves the FilterSearch edge case we have in mind, but seems a little weird for Dropdown? I'm not sure if it makes sense for navigating away from the Dropdown to result in it selecting something. For instance if I had a regular searchbar that used onSelect to trigger searches, I don't think I would want clicking away from the SearchBar to run a search.

I like what https://www.zocdoc.com/ does here (this is one of the examples product used for their FilterSearch vision). If you type a bit into zocdoc's filtersearch, use the arrow keys to highlight an option, and then click away, the query will revert to the little bit you initially typed. 

Might be worth asking product what they want as well though I'm preettttty sure it would be this behavior? The drawback is that if you happened to type out the full name of a filter and then don't select it, it will have the same behavior as now. But in that case the user literally typed out the whole thing so I don't think it's that weird, and would still be my preferred behavior.

Reply:
Actually, `SearchBar` doesn't have anything to handle the `onSelect` event - the `DropdownItem`s in that component  only have an `onClick` event that triggers the search. So if we just write a default `onSelect` handler for the `SearchBar` we should be fine right? Your comment made me realize that I only fixed `FilterSearch`; when a user clicks out of the `SearchBar` dropdown and then hits search, it's still broken.

Reply:
I wouldn't worry about the `SearchBar` for now; I think we may need to talk with Product about their desired behavior on blur for the `SearchBar`. So, we can make another item there if the current `SearchBar` UX is not what they want.

Reply:
Ah yeah when I said "if I had a searchbar that used onSelect" I didn't mean our current searchbar, but a searchbar we or maybe even somebody else implements in the future. For lower level components like Dropdown, we don't want to just make things work in the current world, they also have to make sense with our "mental model" for what a Dropdown is.

Reply:
@tmeyer2115 I think an onBlur makes sense

Reply:
> What is broken about the current `SearchBar`'s behavior?

Currently the `SearchBar` has the same problem that we're fixing with `FilterSearch`, where if you arrow down to select something then click out of the `Dropdown`, the UI shows that it's selected, but when you trigger the search it will still run on the previous query.